src/hooks/xcm/parachainApi/MoonbeamApi.ts

```diff
@@ -0,0 +1,201 @@
+    if (!address) return '0';
+    const symbol = selectedToken.metadata.symbol;
+    const isAstarNativeToken = symbol === 'SDN' || symbol === 'ASTR';
+    try {
+      if (isAstarNativeToken) {
+        // if: SDN or ASTR
+        const addr = await this.getEvmWalletAddress();
+        const tokenAddress = this._AstarTokenId[symbol];
+        const srcChainId = this._EvmId[this._networkName];
+        const balance = await getTokenBal({
+          address: addr,
+          tokenAddress,
+          tokenSymbol: symbol,
+          srcChainId,
+        });
+        return ethers.utils.parseEther(balance).toString();
+      } else {
+        // if: MOVR or GLMR
+        return (await this.getNativeBalance(address)).toString();
+      }
+    } catch (e) {
+      console.error(e);
+      return '0';
+    }
+  }
+
+  public async evmTransferToParachain({
+    toPara,
+    recipientAccountId,
+    amount,
+    selectedToken,
+  }: {
+    toPara: number;
+    recipientAccountId: string;
+    amount: string;
+    selectedToken: Asset;
+  }): Promise<string> {
+    const isConnectedNetwork = await this.isConnectedNetwork();
+    if (!isConnectedNetwork) throw Error('EVM wallet has been connected to the wrong network');
+    if (!this._web3) throw Error('web3 instance is not defined');
+
+    const contract = new this._web3.eth.Contract(moonbeamXcmAbi as AbiItem[], PRE_COMPILED_ADDRESS);
+    const address = await this.getEvmWalletAddress();
+    const [nonce, gasPrice] = await Promise.all([
+      this._web3.eth.getTransactionCount(address),
+      this._web3.eth.getGasPrice(),
+    ]);
+
+    let currencyAddress = '';
+    const symbol = selectedToken.metadata.symbol;
+    const isAstarNativeToken = symbol === 'SDN' || symbol === 'ASTR';
+    if (isAstarNativeToken) {
+      currencyAddress = this._AstarTokenId[symbol];
+    } else {
+      currencyAddress = NATIVE_TOKEN_ADDRESS;
+    }
+
+    const destinationParents = '1';
+    const paraId = String(toPara);
+
+    // Ref: https://docs.moonbeam.network/builders/xcm/xcm-transactor/
+    const parachainId = '0x00000007d' + paraId[paraId.length - 1];
+    const stripPrefixDestAddr = this._web3.utils.stripHexPrefix(
+      u8aToHex(decodeAddress(recipientAccountId))
+    );
+    const destAddress = '0x01' + stripPrefixDestAddr + '00';
+    const destination = [destinationParents, [parachainId, destAddress]];
+    const weight = '4000000000';
+
+    const rawTx: TransactionConfig = {
+      nonce,
+      gasPrice: this._web3.utils.toHex(gasPrice),
+      from: address,
+      to: PRE_COMPILED_ADDRESS,
+      value: '0x0',
+      data: contract.methods.transfer(currencyAddress, amount, destination, weight).encodeABI(),
+    };
+    const estimatedGas = await this._web3.eth.estimateGas(rawTx);
+    const txDetails = await this._web3.eth.sendTransaction({ ...rawTx, gas: estimatedGas });
+
+    const blockMilliSec = 15000;
+    // Memo: wait for syncing in Astar chain
+    await wait(blockMilliSec);
```

Comment:
can we listen the event once astar chain is syncing instead of waiting?

Reply:
I've fixed in this commit. https://github.com/AstarNetwork/astar-apps/pull/471/commits/6ddb9fcfeae65a7aa0a88e0654b78cdf25a6c72e